lib/Doctrine/DBAL/Types/DateIntervalType.php

```diff
@@ -22,7 +22,7 @@ public function getName()
      */
     public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
     {
-        $fieldDeclaration['length'] = 20;
+        $fieldDeclaration['length'] = 255;
```

Comment:
Afaik that only applies to the memory storage type.
This being DBAL the length should not consider only one specific platform. And if I'm not mistaken a developer could adjust the length if needed when defining the schema.
So this is mainly about finding a sensible default length.